ansible-scylla-node/tasks/upgrade/node_verification.yml

```diff
@@ -0,0 +1,65 @@
+---
+
+# Check node status
+- name: Check for leaving nodes
+  ansible.builtin.uri:
+    url: "http://{{ scylla_api_address }}:{{ scylla_api_port }}/storage_service/nodes/leaving"
+    method: GET
+  register: cluster_leaving_nodes
+  until: cluster_leaving_nodes.status == 200
+  retries: "{{ upgrade_api_default_retries |int }}"
+  delay: "{{ upgrade_api_default_delay |int }}"
+
+- name: Check for moving nodes
+  ansible.builtin.uri:
+    url: "http://{{ scylla_api_address }}:{{ scylla_api_port }}/storage_service/nodes/moving"
+    method: GET
+  register: cluster_moving_nodes
+  until: cluster_moving_nodes.status == 200
+  retries: "{{ upgrade_api_default_retries |int }}"
+  delay: "{{ upgrade_api_default_delay |int }}"
+
+- name: Check for joining nodes
+  ansible.builtin.uri:
+    url: "http://{{ scylla_api_address }}:{{ scylla_api_port }}/storage_service/nodes/joining"
+    method: GET
+  register: cluster_joining_nodes
+  until: cluster_joining_nodes.status == 200
+  retries: "{{ upgrade_api_default_retries |int }}"
+  delay: "{{ upgrade_api_default_delay |int }}"
+
+- name: Check for down nodes
```

Comment:
this check will be tricky, since you will have nodes starting and you can only perform it AFTER all nodes are UN
(and in such case this test is moot)
install has a check for this

how big was a cluster that you tested this with? 3 nodes or more?

Reply:
The code copied our official upgrade doc. That's the way it's suggested to do it :/ I've tested it with a 3 nodes cluster